src/core/server/saved_objects/migrations/core/document_migrator.ts

```diff
@@ -257,13 +260,24 @@ function buildDocumentTransform({
   };
 }
 
-function applyMigrations(doc: SavedObjectUnsanitizedDoc, migrations: ActiveMigrations) {
-  while (true) {
-    const prop = nextUnmigratedProp(doc, migrations);
-    if (!prop) {
-      return doc;
+function applyMigrations(
+  doc: SavedObjectUnsanitizedDoc,
+  migrations: ActiveMigrations,
+  log: Logger
+) {
+  try {
+    while (true) {
+      const prop = nextUnmigratedProp(doc, migrations);
+      if (!prop) {
+        return doc;
+      }
+      doc = migrateProp(doc, prop, migrations);
     }
-    doc = migrateProp(doc, prop, migrations);
+  } catch (e) {
+    if (!e.alreadyLogged) {
+      log.error(`Error applying migrations to doc ${doc.type}:${doc.id}: ${e}`);
+    }
```

Comment:
The errors were already logged when thrown from within the migration function, via the `wrapWithTry` wrapper. Only non-logged errors are the one from the actual DocumentMigrator code.

As the migration function errors got some specific infos, I did not remove them when adding this try/catch block. Instead, I use this `alreadyLogged` flag trick to avoid logging the error twice, while preserving the thrown error.